src/wallet/db.cpp

```diff
@@ -71,8 +71,13 @@ BerkeleyEnvironment* GetWalletEnv(const fs::path& wallet_path, std::string& data
         env_directory = wallet_path;
         database_filename = "wallet.dat";
     }
+    // Ensure that the directory does not end with a trailing separator to avoid
+    // creating two Berkeley environments in the same directory
+    while ((env_directory.string().back() == '/') || (env_directory.string().back() == '\\'))
+        env_directory = env_directory.remove_trailing_separator();
```

Comment:
I would prefer the code down there. It might cause an infinity loop on Unix if the wallet filename is `wallet\`
```cpp
while (fs::detail::is_directory_separator(env_directory.string().back())) {
    env_directory.remove_trailing_separator();
}
```

Reply:
Thank you, that's much cleaner, fixed!